support-files/mysql-log-rotate.sh

```diff
@@ -1,37 +1,23 @@
 # This logname can be set in /etc/my.cnf
-# by setting the variable "err-log"
-# in the [safe_mysqld] section as follows:
+# by setting the variable "log_error"
+# in the [mysqld] section as follows:
 #
-# [safe_mysqld]
-# err-log=@localstatedir@/mysqld.log
-#
-# If the root user has a password you have to create a
-# /root/.my.cnf configuration file with the following
-# content:
-#
-# [mysqladmin]
-# password = <secret> 
-# user= root
-#
-# where "<secret>" is the password. 
-#
-# ATTENTION: This /root/.my.cnf should be readable ONLY
-# for root !
+# [mysqld]
+# log_error=@LOG_LOCATION@
 
-@localstatedir@/mysqld.log {
-        # create 600 mysql mysql
+@LOG_LOCATION@ {
+        create 600 mysql mysql
         notifempty
 	daily
         rotate 3
         missingok
         compress
     postrotate
 	# just if mysqld is really running
-	if test -x @bindir@/mysqladmin && \
-	   @bindir@/mysqladmin ping &>/dev/null
-	then
-	   @bindir@/mysqladmin --local flush-error-log \
-              flush-engine-log flush-general-log flush-slow-log
-	fi
+        if [ -e @PID_FILE_DIR@/@DAEMON_NO_PREFIX@.pid ]
```

Comment:
There's no default ```@PID_FILE_DIR@``` or  ```@DAEMON_NO_PREFIX@```. For systemd there may be no pid file either so maybe 'ExecReload' in the service file?

Going to a SIGHUP is a regression on MDEV-11610 which reduces impact of the rotate (at a time when binlog rotates where causing segfaults). I think I suggested in a MDEV somewhere that having a different signal for this minimal logrotation which didn't immediately get rejected.

Eventually I wouldn't mind seeing a mysql unix socket user installed and then run the logrotate script under the mysql user.

Reply:
pidfile via ```my_print_defaults --mysqld``` maybe.